src/callback_handler/kubernetes/reflector.rs

```diff
@@ -23,12 +23,21 @@ where
     K::DynamicType: Eq + Hash + Clone,
     W: Stream<Item = watcher::Result<watcher::Event<K>>>,
 {
-    stream.inspect_ok(move |event| {
-        if let Err(err) = last_change_seen_at.send(Instant::now()) {
-            warn!(error = ?err, "failed to set last_change_seen_at");
-        }
-        writer.apply_watcher_event(event)
-    })
+    stream
+        .take_while(|event| {
+            if let Err(watcher::Error::InitialListFailed(err)) = event {
```

Comment:
Maybe add a couple of lines explaining why we care to handle only this specific error type